src/error.rs

```diff
@@ -9,15 +9,23 @@ pub struct Error {
     source: Option<Source>,
 }
 
-#[derive(Debug)]
+#[derive(Debug, Eq, PartialEq)]
 pub(crate) enum ErrorKind {
```

Comment:
one thing to think about is whether or not we want to split error kind into different enums as the number of errors grows.

Reply:
Probably